code/game/objects/items/weapons/mop.dm

```diff
@@ -42,7 +42,7 @@
 
 	var/turf/T = get_turf(A)
 
-	if(istype(A, /obj/item/weapon/reagent_containers/glass/bucket))
+	if(istype(A, /obj/item/weapon/reagent_containers/glass/bucket | /obj/structure/janitorialcart))
```

Comment:
Apparently this work, but we aren't sure why or if it will always work correctly. If its consistent it's a pretty cool way to compress long if istype() chains a bit. It'd be good if someone more versed in binary operators could give some insight here.